kubernetes-kit-starter/src/test/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/SerializationDeserializationTest.java

```diff
@@ -72,6 +86,81 @@ void processComponentWithNullTransients() throws Exception {
+        TestConfig.UnserializableWrapper wrapper = new TestConfig.UnserializableWrapper(
+                new TestConfig.Unserializable("Unserializable"));
+
+        MockedUI ui = new MockedUI();
+        ComponentUtil.setData(ui, "WRAPPER", wrapper);
+
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        Assertions
+                .assertThatExceptionOfType(
+                        UnserializableComponentWrapperFoundException.class)
+                .isThrownBy(() -> TransientInjectableObjectOutputStream
+                        .newInstance(os, handler).writeWithTransients(ui));
+    }
+
+    @Test
+    void detachedUnserializableWrapper_unlockedVaadinSession_throws()
+            throws Exception {
+        TestConfig.UnserializableWrapper wrapper = new TestConfig.UnserializableWrapper(
+                new TestConfig.Unserializable("Unserializable"));
+
+        MockedUI ui = new MockedUI();
+        VaadinSession session = mock(VaadinSession.class);
+        when(session.hasLock()).thenReturn(false);
+        ui.getInternals().setSession(session);
+        ComponentUtil.setData(ui, "WRAPPER", wrapper);
+
+        CurrentInstance.clearAll();
+        VaadinSession.setCurrent(session);
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        try {
+            Assertions
+                    .assertThatExceptionOfType(
+                            UnserializableComponentWrapperFoundException.class)
+                    .isThrownBy(() -> TransientInjectableObjectOutputStream
+                            .newInstance(os, handler).writeWithTransients(ui));
+        } finally {
+            CurrentInstance.clearAll();
+        }
+    }
+
+    @Test
+    void detachedUnserializableWrapper_lockedVaadinSession_handlesUnserializable()
+            throws Exception {
+        TestConfig.UnserializableWrapper wrapper = new TestConfig.UnserializableWrapper(
+                new TestConfig.Unserializable("Unserializable"));
+
+        MockedUI ui = new MockedUI();
+        VaadinSession session = mock(VaadinSession.class);
+        ReflectionTestUtils.setField(session, "requestHandlers",
+                new LinkedList<>());
+        ReflectionTestUtils.setField(session, "destroyListeners",
+                new CopyOnWriteArrayList<>());
+        when(session.hasLock()).thenReturn(true);
+        ui.getInternals().setSession(session);
+        ComponentUtil.setData(ui, "WRAPPER", wrapper);
+
+        CurrentInstance.clearAll();
+        VaadinSession.setCurrent(session);
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        try (var mockedStatic = mockStatic(
+                UnserializableComponentWrapper.class)) {
+            TransientInjectableObjectOutputStream.newInstance(os, handler)
+                    .writeWithTransients(ui);
+            mockedStatic.verify(() -> UnserializableComponentWrapper
+                    .beforeSerialization(wrapper), times(2));
```

Comment:
Is it invoked twice because of the reason mentioned in https://github.com/vaadin/kubernetes-kit/pull/183/files#r2046345226?